control-plane/agents/src/bin/core/controller/io_engine/v1/translation.rs

```diff
@@ -208,6 +208,7 @@ impl TryIoEngineToAgent for v1::replica::ReplicaSnapshot {
                 .timestamp
                 .clone()
                 .and_then(|t| std::time::SystemTime::try_from(t).ok())
+                // only valid on create, not on list! should we handle this as None ?
```

Comment:
github doesn't allow to comment on unchanged lines :\

Should the ResourceKind above be modified to Resourcekind::ReplicaSnapshot as you have added new kind now.

```
SnapshotId::try_from(self.snapshot_uuid.as_str()).map_err(|_| {
                SvcError::InvalidUuid {
                    uuid: self.snapshot_uuid.to_owned(),
                    kind: ResourceKind::VolumeSnapshot,
                }
            })?,
```

Reply:
Yeah I know, it has some limitations..
Done :)